app/controllers/your-psp/your-psp-tasks.service.test.js

```diff
@@ -0,0 +1,153 @@
+const { expect } = require('chai')
+const { getCurrentCredential } = require('../../utils/credentials')
+const { getStripeTaskList, stripeTaskListIsComplete } = require('./your-psp-tasks.service')
+const gatewayAccountFixtures = require('../../../test/fixtures/gateway-account.fixtures')
+
+describe('Progress of Stripe PSP tasklist page', () => {
```

Comment:
Every test is checking multiple methods `getStripeTaskList` and `stripeTaskListIsComplete`.

Would be good to have a separate group of tests so it is clear what is being tested. So brought structure could be like below.

```
describe('Stripe task list', () => {

describe('Get Stripe task list status', () => {
   it('should calculate Stripe status (enabled & completed) correctly for each task when completed', () => {
   })
   it('should calculate Stripe status (enabled & completed) correctly for each task when not completed', () => {
   })
   it('should return upload govenment document as enabled only when all the other tasks are completed', () => {
   })
   it('should throw error if payment provider is not Stripe', () => {
   })
})

describe('Calculate Stripe tasks completess', () => {
   it('should return true if all tasks are completed', () => {
   })
   it('should return false if none of the tasks are completed', () => {
   })
   it('should return false if any task is not completed', () => {
   })
})

})
```

Reply:
Yes this seems a better approach.